rpc/watchconfirm.go

```diff
@@ -0,0 +1,109 @@
+package rpc
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"github.com/Qitmeer/qitmeer/common/hash"
+	"github.com/Qitmeer/qitmeer/core/blockchain"
+	"github.com/Qitmeer/qitmeer/core/types"
+	"github.com/Qitmeer/qitmeer/rpc/client/cmds"
+)
+
+type TxConfirm struct {
+	Confirms uint64
+	TxHash   string
+}
+
+func (w *WatchTxConfirmServer) AddTxConfirms(confirm TxConfirm) {
+	if w == nil {
+		w = &WatchTxConfirmServer{}
+	}
+	if _, ok := (*w)[confirm.TxHash]; !ok {
+		(*w)[confirm.TxHash] = TxConfirm{}
+	}
+	(*w)[confirm.TxHash] = confirm
+}
+
+type WatchTxConfirmServer map[string]TxConfirm
+
+func (w *WatchTxConfirmServer) Handle(wsc *wsClient) {
+	if w == nil || len(*w) <= 0 {
+		return
+	}
+	bc := wsc.server.BC
+	txIndex := wsc.server.TxIndex
+	if txIndex == nil {
+		log.Error("specify --txindex in configuration")
+	}
+	for tx, txconf := range *w {
+		txHash := hash.MustHexToDecodedHash(tx)
+		blockRegion, err := txIndex.TxBlockRegion(txHash)
+		if err != nil {
+			log.Error(err.Error(), "txhash", txHash)
+			continue
+		}
+		if blockRegion == nil {
+			if bc.CacheInvalidTx {
```

Comment:
Your purpose is just to confirm the transaction, why not `continue` directly?  I don't think it's necessary to deal with all the illegal transactions.